consensus/poet/core/tests/test_consensus/test_poet_fork_resolver.py

```diff
@@ -0,0 +1,656 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+import shutil
+import tempfile
+
+from importlib import reload
+
+from unittest import TestCase
+from unittest import mock
+
+from sawtooth_poet.poet_consensus import poet_fork_resolver
+
+from sawtooth_poet_common.protobuf.validator_registry_pb2 \
+    import ValidatorInfo
+from sawtooth_poet_common.protobuf.validator_registry_pb2 \
+    import SignUpInfo
+
+
+# @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.BlockWrapper')
+# @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.PoetConfigView')
+class TestPoetForkResolver(TestCase):
+
+    def setUp(self):
+        # pylint: disable=invalid-name,global-statement
+        global poet_fork_resolver
+        # PoetBLockResolver uses class variables to hold state
+        # so the module needs to be reloaded after each test to clear state
+        poet_fork_resolver = reload(poet_fork_resolver)
+
+        self._temp_dir = tempfile.mkdtemp()
+
+    def tearDown(self):
+        shutil.rmtree(self._temp_dir)
+
+    @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.'
+                'ConsensusStateStore')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.factory')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.'
+                'ConsensusState')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.'
+                'ValidatorRegistryView')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.utils')
+    def test_new_fork_head_not_poet_block(
+            self,
+            mock_utils,
+            mock_validator_registry_view,
+            mock_consensus_state,
+            mock_poet_enclave_factory,
+            mock_consensus_state_store):
+
+        """ Test verifies that if the new fork head is not a valid block,
+            raises appropriate exception
+        """
+
+        # create a mock_validator_registry_view
+        mock_validator_registry_view.return_value.get_validator_info. \
+            return_value = \
+            ValidatorInfo(
+                name='validator_001',
+                id='validator_deadbeef',
+                signup_info=SignUpInfo(
+                    poet_public_key='00112233445566778899aabbccddeeff'))
+
+        # Make utils pretend it cannot deserialize the wait certificate
+        # of the new fork head
+        mock_utils.deserialize_wait_certificate.return_value = None
+
+        mock_block_cache = mock.MagicMock()
+        mock_state_view_factory = mock.Mock()
+
+        # create mock_poet_enclave_module
+        mock_poet_enclave_module = mock.Mock()
+        mock_poet_enclave_factory.get_poet_enclave_module.\
+            return_value = mock_poet_enclave_module
```

Comment:
I don't think lines 83-86 are necessary.  Since you are not actually faking out any of the enclave functions, I don't think that assigning a `Mock` object to `mock_poet_enclave_factory.get_poet_enclave_module.return_value` is doing anything that the mock isn't already automatically doing.  However I may be wrong.